### DISCUSSION TITLE
equipment needed - new to GenMon

### DISCUSSION THREAD

**Discussion by kjwilliams1974** [jgyates/genmon#1073] [General]
Hello guys - donation sent - getting ready to install GenMon.  Setup is Generac 38KW liquid cooled.  Not sure if Evolution 1 or 2 controller.  Have a Pi3 laying around - new in box.  Want to order all cables, etc. pre-made along with HAT.  What exactly do I need.  It is not clear when trying to order equipment, as the HAT ordering source listed is asking If I have an Evolution 1 or 2 controller.  According to my sources at GenTek, all liquid cooled units are just Evoltions, not 1 or 2 and the ver 1 or 2 are for air cooled units... So, I am a little confused on that aspect.  The other question I have is powering the Pi.  Are we able to power the unit from the controller pre-made cables?  I am in the hot / humid south Texas area too.  Do I need the optional fan to keep the Pi from overheating....

**Comment by jgyates** [comment]
Yes, it is just an Evoltuion. No 2.0 exist. You can just select  Liquid Cooled Evolution 1.0. I don't think the Evolution 2.0 Liquid cooled controllers exist (yet).

A Pi on the fan is always a good idea, especially in the Tx heat with a generator running nearby.

I will let @skipfire answer the other questions as he is the pintsize.me person in the know.

**Reply by kjwilliams1974** [reply]
thank you!

**Reply by skipfire** [reply]
Pick the Liquid Cooled Evolution 1.0, the hardware is the same I just have the controller question in place so that I have that information should someone request support from me since there are differences in capabilities of the controllers, but if it's wrong then no harm done and it can be sorted out while working on support.  The pre-loaded unit has the OS and GenMon already installed; you then configure your WiFi details, do security stuff (changing the password), set the timezone, and configure GenMon the way you want it (setting up notifications if you want that).

All the HATs/pHATs come with a data cable and a power cable (battery loops to the barrel plug on the HAT) and will power the HAT and the battery.  I don't recommend powering any Pi from the controller directly because if it draws too much power it temporarily throws off the voltage sensors making the generator think there is a low utility voltage condition and it starts the generator and transfers power.

I am in southeastern Louisiana (just north of New Orleans) so I get the heat and I have not had overheating issues with my Pi 3A+ (which is what's in the pre-loaded units I have left, in a week I expect to have some Pi Zero 2 preloaded units) in the generator enclosure even after Ida when I was on generator for 6 days until the generator shut down and airflow stopped and the ambient spiked over 85c due to radiant heat from the engine and no fan will help when the ambient is already above the limit.  This past summer I also had no problem even with the run of 100+ days we had.  It does get warm, but it didn't get critically warm.  Of course the position of your generator, how much sun hits it through what hours of the day, etc, can all change your experience.  The enclosure does really kill wifi strength, so many users place their setup in weatherproof boxes on the outside of (but attached to) the generator, which may offer you an option where the sun doesn't hit it during the hottest/brightest hours.  I moved mine outside because of how frequently I have to access it and have it in a non-weatherproof enclosure, but the gap is on the bottom so unless I decide to get a sprinkler system and aim it right at the spot it will stay dry.  I have my cables just running under one of the enclosure panels and it's only exposed for about 3 inches before it goes into my plastic box.

If you have more questions or if I missed something or wasn't clear please reply.

**Reply by kjwilliams1974** [reply]
thanks!  Trying to figure out which way to go.  I have the Pi3 sitting here - new in box.  Was a gift...  Generator is next to my well house - wifi or cat6 is not an issue.  Robust Enterprise level wifi network inside home and outside for the property.  Could power off of POE too, as I have switch in the well house that has a few open ports...  Any input?

**Reply by jgyates** [reply]
In general I recommend wired over wireless for the modbus traffic (either Ethernet based serial over TCP or direct RS-232 serial). Number 16 on the [known issues wiki page](https://github.com/jgyates/genmon/wiki/Appendix-D-Known-Issues#16-communication-errors) has more info on the limits of wifi with modbus.

Wifi for the web interface is fine.

Personally I like PoE for the convenience of cycling power if needed as you can typically do this from a PoE based switch.

This partly depends on where you mount your pi, but this product has been used with some success:

https://www.waveshare.com/rs232-to-eth-b.htm

this allows you to run cat6 to the controller and just mount this device, then the pi is inside.

Alternatively you can mount the pi in your generator housing and do a direct connection serial. Then the pi connects to your local network with wifi. If the pi has an issue you may have to power cycle by going outside, but that is not a show stopper either.

**Reply by skipfire** [reply]
I also have solid wifi in and out, but the generator enclosure is basically a faraday cage and my tools tell me it reduced WiFi strength by 20dB which is 99% loss in WiFi power which may not be a problem for the WAP but could be for the Pi's transmitter.  Ethernet is a great option if you can do it, for most it's not a realistic or cost effective option.  PoE for powering can work if the Pi 3 is a 3B+ which is when it was first introduced, the 3A+ and 3B do not support PoE (3A+ doesn't even have an Ethernet port).  If you do go the PoE route you have to find a PoE HAT that is designed to be in the middle or has a passthrough so you can put extra long pins through it.  The PoE HATs require the extra 4-pin header so it has to be the HAT closest to the Pi.

The only other question is how comfortable are you with doing the setup of the OS and software with provided command line commands?  I do also sell the micro-SD cards from the pre-loaded units by itself if you want to use your Pi but not deal with setup of the OS and install of the software.

**Reply by skipfire** [reply]
And I'll agree with @jgyates on the rs232 to ethernet option, it's a very viable and cheap path and then you don't buy anything from me.

**Reply by kjwilliams1974** [reply]
I am not concerned about $.  I just want the best of all the options.  With that being said, let's move on the best solution...  I would like to mount the PI in the well house and not in the generator or outside the generator in weather proof box.  The well house is controlled and has back up UPS, etc and much more weather friendly...

**Reply by skipfire** [reply]
How close is the generator to the well house?  My guess is that it is far enough that you will have to look at alternate cabling options (the longest Molex I've seen available is 1 meter), in which case you may as well put the Pi inside your house.  From what you have said so far and a desire to not have the Pi in/at the generator, I think the best option would be the rs232 to ethernet adapter in the generator with the ethernet line going to your switch, then put your Pi close to your switch and I'm guessing you have a UPS there too since you have one in the well house.  The only drawback to the rs232 adapter is the for data and power, but that's not a very large effort to overcome.

**Reply by kjwilliams1974** [reply]
@skipfire - thanks for the input - Gen is about 10 to 12 feet from the well house.  What do you all think about conduit to the generator, shielded cat6 (what I have in my possession) to the ethernet to RS-232 port adapter to the controller?  I could just use the POE and open port.  The Pi that I have is a Pi3B.  No POE then, right?  I have power.  I may just order one completely configured from you... that would be the best solution...

**Reply by skipfire** [reply]
Given your setup, I would suggest the plan you described using the RS232 in the generator with an ethernet cable through a conduit to get it on your network, and then stick your Pi wherever your network equipment is (or wherever you prefer) and it communicates over the network with the RS232 adapter.  It lets the Pi not be at the generator and has everything hardwired.  This is probably the path I'd go if I could easily get ethernet to my generator.  I'm sure someone is going to read this and think I'm a terrible salesperson recommending something that isn't my products.

You are correct that the Pi 3B does not support PoE.  As for ordering one from me, I currently only have Pi 3A+ which does not have an Ethernet port so it would have to be WiFi or you would have to add a USB Ethernet adapter.  If you want to stick with Ethernet you can buy the Standard HAT and a pre-loaded micro-SD card from me [Misc Parts](https://pintsize.me/store/ols/products/miscellaneous-parts), and use your Pi and you end up with the same basic setup as my pre-loaded, you just have to physically assemble yourself which is just lining up the HAT and doing the screws & nuts.  I do also sell a Molex to RJ45 kit (IT IS NOT A NETWORK COMPATIBLE DEVICE), but it does use standard ethernet cables to do extended runs.  You could then place one part of that kit in the generator, and the other end in your house connecting to the generator HAT.  With that path you could have the Pi inside and the sequence of equipment is Evolution -> Molex cable 1 -> Molex to RJ45 adapter -> 8 conductor wire with RJ45 -> RJ45 to Molex adapter -> Molex cable 2 -> HAT (or Molex USB that I sell) -> Pi.

**Reply by kjwilliams1974** [reply]
Thanks - have to fly in the morning early. I will take a look at this again and consider what I need...  Thank you very much for your time and energy! Let me clarify - can I use your cable in place of the adapter or do I need it too? I think the route that I want to go is like you described above - RS232 from controller over ethernet to HAT or USB to PI3B using your pre-loaded card and whatever else I need... I will put the PI in a box even though it is in the well house with a fan - as it does get hot in the well house... Can someone advise on PI fan?  If you can, I just need a list of things I need to to order and since you have helped me so much, I will order from you too... and again, THANK U!

**Comment by kjwilliams1974** [comment]
Ok,  back up…. I think I am going to go the route of ordering a pre-loaded card, the HAT for the pi 3 that I have and the hardware to change from 232 to rj45.  I need to go back to rs232 for connection to the hat…. Any help on the pieces needed?

**Reply by skipfire** [reply]
I do sell a Molex to RJ45 kit, if you used that so then you would just need to look at a [RS232 adapter](https://www.amazon.com/gp/product/B07S7M24C3/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&th=1) (check if you need male or female) and grab one of the [Molex extensions](https://www.amazon.com/gp/product/B000M802RG/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1) to cut up to connect to the RS232 adapter.  That would get you from DB9 to Molex, and from the Molex point you can use the Molex to RJ45 kit with a Cat5/6 cable in between to run the serial over a greater distance.  The HAT and the RJ45 kit each come with a Molex cable (1 meter with the HAT, 6" with the Molex RJ45 kit), but you will have to locally supply power to the Pi from a UPS.

However, I was under the impression that the Evolution on Liquid Cooled had a Molex connector not a DB9, if you have a DB9 on the controller face you may have an H-100 which I believe has extra setup steps that I am not aware of.

**Reply by kjwilliams1974** [reply]
Thx.  I am not sure what the liquid cooled unit has other than the evolution controller.  Anyone ?

**Reply by skipfire** [reply]
Can you post a picture of the controller?  That really help identifying what it is.  Please note, pictures have to be posted through the web, attachments to an email do not work.

**Reply by kjwilliams1974** [reply]
I hate to be a pain, but I am a little lost at what all it is that I need for sure.  Molex to RJ 45 - can’t locate on your listings….

**Comment by jgyates** [comment]
my liquid cooled controller is an Evolution and has an 8 pin molex connector

**Comment by kjwilliams1974** [comment]
38kw liquid cooled Generac with evolution controller.  Pi3 will be in well house adjacent to the generator.  Planning on running cat6 to adapters in generator to connect  with pi and hat inside well house, connected to a Poe switch…. I am just a little lost on exactly what it is that I need.  I have ordered from AMZ, the RS232 to RJ45 adapter and the molex extension.  I will need a hat, preloaded card and a molex to RJ-45 kit…. Anything else?  Thx in advance for your help and patience…

**Reply by skipfire** [reply]
On the listing for the HATs there is a Splitter/Injector dropdown with a `RJ45 Splitter/Adapter` option, that is the Molex to RJ45 equipment which comes in 2 parts.  On the generator end it has a pair of Molex ports and a RJ45 port (each labeled for the appropriate connection), the other device has a RJ45 port and a single Molex for the Pi/HAT end of things.  As a reminder, the Pi 3A+ that is in the pre-loaded units does not have a network port, it is WiFi only.

Based on the latest stuff you have posted, from the [Generator HATs and pHAT](https://pintsize.me/store/ols/products/generator-hats-and-phat/v/PGI-RPL) page you will want a `Pre-loaded v2.6+ Pi 3A+` and a Splitter/Injector option of `RJ45 Splitter/Adapter` - you will have to provide your own Cat5/6 cable, RS232 spec allows up to 50 feet total, I've tested up to 29 feet (the longest I had was 25 feet, plus the Molex cables was within a few inches of 29 feet).  Add the RS232 adapter and the short Molex cable from Amazon to get you from DB9 to Molex on the generator end and you should be good to go.

**Comment by dfxinc** [comment]
Did you ever get this installed on your 38kw?
I have a 2023 32kw liquid cooled with same controller, but the controller has 5 separate connectors...any help with where and how you connected would be appreciated. 

Im on controller FW 1.52...any issues with encryption on this version?

Photos:
![IMG_4534](https://github.com/user-attachments/assets/930a8fb8-1a00-4aa3-83e8-0ca09596a937)
![IMG_4540](https://github.com/user-attachments/assets/ceb054fd-2511-4605-8d98-42c0c6720f45)
![IMG_4531](https://github.com/user-attachments/assets/d6832bc5-7aaf-4bf4-9fc2-175a9373b621)

**Reply by jgyates** [reply]
Firmware V1.52 for liquid cooled works fine and is unencrypted. This is the version of the FW that I have in my controller.

**Reply by jgyates** [reply]
also, mine is an older unit (9 years old) but the controller was replaced. My unit did not come with mobile link and only has one molex connector this size.

**Reply by dfxinc** [reply]
Thanks! are all 8 wired on the molex? Mine appears to only have 6...im pulling it out tonight to take a closer look. Do you have any photos?

**Reply by jgyates** [reply]
The number of wires in the cable that is plugged into the 8 pin molex varies from model to model. For example in mine, there was nothing connected (since it did not come with Mobile Link preinstalled).  My controller only has one 8 pin molex plug, but some others have 2 8 pin plugs. Yes, the cable connected into the 8 pin molex socket typically does not have all 8 pins wired into the plug. The pinout for the 8 pin molex is here:

https://github.com/jgyates/genmon/wiki/3.1--Making-a-Cable

**Reply by jgyates** [reply]
It has been a while since I have taken pics. My initial pictures are in the repository [here](https://github.com/jgyates/genmon/tree/master/Diagrams) but there is not a good shot of the back. I would need to unscrews the controller to get back there on my units. I have a 48kW diesel from 2014.

**Comment by dfxinc** [comment]
No worries, im going to try and take mine apart so I can see exactly what I have

**Reply by dfxinc** [reply]
update...found the molex...its on the far side. The entire controller must be removed.
![IMG_4766](https://github.com/user-attachments/assets/7272ec8b-b440-48f0-b62b-fe32bbf4c717)

Id love to keep MobilLink active as well...but not sure its possible

**Reply by jgyates** [reply]
Great pic, that will likely help others. thanks for posting this

**Reply by skipfire** [reply]
Unfortunately 387 & 388 are RS232 (which only supports one accessory) and I can see the 388 label on the top wire.

**Reply by dfxinc** [reply]
Thanks. I'll post better images when I do the full install in hopes someone else can benefit. 
and yes @skipfire thats what I was afraid of.

BTW- The molex port has a rubber cover labeled "Accessory Port" in case anyone else is looking for it and dont have Mobile Link installed.

**Reply by heffneil** [reply]
Did this work for you?  For some reason I think the pins are wrong on the pihat relative to the plug on this unit.  I am curious what you used and if you had success?

**Reply by dfxinc** [reply]
I still haven't installed mine...Im waiting until I hit 25 hrs and get the first service done. But I see you have the newer liquid cooled unit with Generac 4.5L...I have the previous generation with Mitsubishi engine. I wonder if they modified something in latest model. What is your Controller FW version?

**Reply by heffneil** [reply]
It took us 2 days but the serial port baud rate is 115200 not 9600.  Its the only one and its awful that it was making me pull my eyebrow hairs out (I am bald) so just be aware.

**Reply by jgyates** [reply]
FYI, the baud rate is 115200 but only for the newer 4.5L Liquid cooled engines. All others are 9600 which is the default for genmon.

**Reply by heffneil** [reply]
hahah - I feel like this needs to be made super bold or something.  I would believe that there arne't that many of them but its a small detail that REALLY matters. Maybe adding to the code to ask engine size to adjust automatically?

**Reply by jgyates** [reply]
I put this info in the [known issues wiki](https://github.com/jgyates/genmon/wiki/Appendix-D-Known-Issues) and put a link to the known issues in the discussion questions section check list that pops up when you create a new discussion. 

There are over 1000 discussion thread and over 1000 issue threads and this issue has come up twice so far.

I will consider putting this in the install script, but considering the low numbers of people using 4.5L and the high numbers of people how use air cooled  units and older non 4.5L liquid cooled units, the odds are that this would cause more people entering the in incorrect answer to the question "do you have a 4.5L engine" is probably greater than those who would benefit (i.e this may cause more problems that it fixes). If I get more people running into this issue that will certainly help improve the cost/benefit of this feature.

**Reply by heffneil** [reply]
I agree! its just so obscure its hard to identify.  Like I said its a unique engine / generator size I guess for people looking to install genmon.  I appreciate all the help but it was a hard issue to stomp out! Thanks again!

**Reply by skipfire** [reply]
@jgyates might be a good prominent note in the serial troubleshooting document as well.

**Reply by jgyates** [reply]
that is a good idea about he serial troubleshooting guide. I will add it there.

**Reply by jgyates** [reply]
I updated the serial troubleshooting wiki with a link to this info: https://github.com/jgyates/genmon/wiki/3.6---Serial-Troubleshooting

**Comment by ricracrum** [comment]
Hey, @skipfire 

I purchased a kit from you for my older 20KW Generac used in Florida quite a while ago, and it's been working great. It's a typical install with the Pi at the generator using wifi. Also, I have contributed to the project again, thanks @jgyates!

I have a new generator (Generac 26KW Evolution 2.0 Controller PIN 10000003275) at a house in North Carolina and could use your help piecing together the parts to purchase from you. For this case, I already have a CAT6 line to the generator, but as with the OP here I want to remote the Pi in my IT closet. I have a 1U rack mount that holds 4 Pi devices and will be using one slot for a Pi dedicated to Home Assistant. I'd like to mount a device from you in the same rack. I like the idea of using your Molex to RJ45 in the generator enclosure to my IT closet, then into the Pi, and then connect the Pi to my switch. Ideally, if you have a Pi device I can power using a PoE+ enabled port on the switch, that would be the route I'd prefer for simplicity, but using a power block is okay. 

Thanks!

**Reply by skipfire** [reply]
What I'd recommend is the RJ45 kit to go the distance and then use the USB Molex adapter which works with both Pi and non-Pi solutions.  You can then get a PoE HAT to power whichever Pi model you choose.  I haven't seen HA take a ton of resources (unless you get into the voice & AI stuff), so you may be able to run GenMon on the same Pi as HA, though I have no experience in doing so.  Unfortunately my directly rackmounted product is still 2 or 3 months away from being ready (at best).

If you want to talk about customizing the Kit for your needs hit me up through our website or on [Facebook messenger](https://www.facebook.com/profile.php?id=61555703325298).

**Reply by ricracrum** [reply]
Thanks @skipfire I wanted to start this question here to help anyone else finding this thread seeking the same solution. I'll move this to email on your website for details.

**Comment by JDogg016** [comment]
I have a generator setup similar to @dfxinc, but went the route of buying the RPI and the Hat.
I managed to also find the accessory port behind the controller, but I see that power is fed to the WiFi Module by another wire.

Here is my question:  How did you power the RPI?  Unlike some other videos, I do not know where to look to find the battery and I do not know whether running the wire to the battery is optimal solution.

Any advice would be greatly appreciated.

**Reply by heffneil** [reply]
i bought his kit that came with terminals that allowed you to power off the battery.  I have since switched and not I power the PI from a POE splitter.  I do this with a lot of raspberry PI's.  The poe spliiter will give you a USB C (there are micro usb versions) but they can be had on amazon for less than $20.

**Reply by skipfire** [reply]
Motorcycle USB adapters with a good cable are a solid choice.  One example is [https://www.amazon.com/dp/B0CQR7JRRH](https://www.amazon.com/dp/B0CQR7JRRH), but be sure to check the amperage/wattage depending on which Pi you get. Any will likely work with Pi Zero 2 or PI 3s, but if you are tryin got run a Pi 4 or Pi 5 (be aware of thermal issues with those) you may need one that outputs more power.
